cmd/upgrade/app/v1alpha1/executor.go

```diff
@@ -40,26 +49,71 @@ type ExecutorBuilder struct {
 //It adds object in ExecutorBuilder struct with the help of config
 func ExecutorBuilderForConfig(cfg *apis.UpgradeConfig) (b *ExecutorBuilder) {
 	b = &ExecutorBuilder{}
+
+	instanceName := os.Getenv(instanceNameENVKey)
+	if instanceName == "" {
+		b.errors = append(b.errors,
+			errors.Errorf("failed to instantiate executor builder: %s ENV not present", instanceNameENVKey))
+		return
+	}
+	instanceNamespace := os.Getenv(instanceNamespaceENVKey)
+	if instanceNamespace == "" {
+		b.errors = append(b.errors,
+			errors.Errorf("failed to instantiate executor builder: %s ENV not present", instanceNamespaceENVKey))
+		return
+
+	}
+	instanceUID := types.UID(os.Getenv(instanceUIDENVKey))
+	if instanceUID == "" {
+		b.errors = append(b.errors,
+			errors.Errorf("failed to instantiate executor builder: %s ENV not present", instanceUIDENVKey))
+		return
+
+	}
+
 	castObj, err := cast.KubeClient().
 		Get(cfg.CASTemplate, metav1.GetOptions{})
 	if err != nil {
 		b.errors = append(b.errors,
-			errors.WithMessagef(err,
+			errors.Wrapf(err,
 				"failed to instantiate executor builder: %s", cfg))
 		return
 	}
 
+	tasks := []apis.UpgradeResultTask{}
+	for _, taskName := range castObj.Spec.RunTasks.Tasks {
+		task := apis.UpgradeResultTask{
+			Name: taskName,
+		}
+		tasks = append(tasks, task)
+	}
+
 	engines := []cast.Interface{}
 	for _, resource := range cfg.Resources {
 		resource := resource // pin it
+		upgradeResult, err := NewUpgradeResultBuilder().
```

Comment:
We need to write in comments about what is happening. e.g. a upgrade result is created per resource that needs to be upgrade.